### DISCUSSION TITLE
With the immich duplicates utility - any way to prioritise HEIC format ?

### DISCUSSION THREAD

**Discussion by vw-kombi** [immich-app/immich#10725] [General]
Pretty much what the title says.  We are all iphones, and due to a time before we got icloud, my workflow was connecting the phones and the app I used conferted HEIC to JPG on my PC.  Now I imported the files, the icloud content and google photos - and I have 21,000 dupes - ad most seem to be a copy of JPG and HEIC taken on same time.   The duplicate utility default to the JPG to keep - as I guess it is larger ?  but I would like to keep the HEIC.  Any way to set that ?  And if so, any way to batch accept all of them ?

**Comment by stevenmcastano** [comment]
This isn't actually about immich itself... but I've been having the exact same problem and I got a bit tired of waiting! So... and let me preface this buy saying I'm not a developer... at all... but I've been giving it my best shot!

I've been working on messing around with an app called [Immich Duplicate Finder](https://github.com/vale46n1/immich_duplicate_finder) which was created by @vale46n1 and I've been loving it so far... check it out, it's not exactly the same... no stacking option, and it only compares 2 images at a time... but I've been hacking away on the python code, one of the only languages I know decently well... and I've gotten it to the point there it will detect if there's an HEIC and it's extracted JPG and show you in the UI so you can better pick which one to toss.

No auto delete yet, not much in the way of config yet... but I'm getting there. If anyone would be willing to help... so show me how to contribute similar code to this project, I'd be happy to pitch in more... and if it's something you're interesting in testing, I'll do what I can to package up the pretty hacky code and get a fork posted!

Normally it highlight differences in a color with green being the "better" one. When it finds HEICs and JPGs and notices they are dups, it usually only highlights the size and makes the larger one green thinking its got more data so it must be better even though everything else matches.

This is the original output from the code in the current repo:

![Screenshot 2025-02-06 at 3 51 15 AM](https://github.com/user-attachments/assets/c3a7139d-43cd-47f6-af08-826c92f72498)

Now, with my new code you can see I added the "GPS Data" section to you can tell if you're delete a copy with GPS data when the other one doesn't have any, I also added the "Camera Info" sections which it puts together from EXIF data in the image, not just the "lens" tag... then the big win for me is that it now can tell if it's an HEIC pair, colors the HEIC file green, make the JPG file red and changes the size coloring to yellow if they're different so you can still see there's a difference, but you'd probably still want to go with the HEIC anyway. Here's what that looks like:

![Screenshot 2025-02-06 at 3 30 40 AM](https://github.com/user-attachments/assets/9cb9bb22-905f-438b-8165-8b2d35a9486f)

P.S. The "EXT_DIFF_HEIC" text can be turned off... that's just running in a debug mode so I can more easily find examples while testing!
